--- a/configure.ac
+++ b/configure.ac
@@ -411,9 +411,12 @@
 		cp include/pscom.h "${TOP_BUILDDIR}/pscom"
 	],[with_pscom_allin=$with_pscom_allin; TOP_BUILDDIR=$TOP_BUILDDIR])
 
-	AC_CHECK_FILE(["${PSCOM_ALLIN_DIR}/lib/all/pscom_all.c"], [
-		PSCOM_ALLIN_DIR="${PSCOM_ALLIN_DIR}/lib/all/"
-	])
+	dnl Use shell test instead of AC_CHECK_FILE to support cross-compilation
+        dnl (the checked file never exists in the --with-pscom-allin source bundle anyway)
+        if test -r "${PSCOM_ALLIN_DIR}/lib/all/pscom_all.c" ; then
+                PSCOM_ALLIN_DIR="${PSCOM_ALLIN_DIR}/lib/all/"
+        fi
+
 	export MPICHLIB_CFLAGS="${MPICHLIB_CFLAGS} -DPSCOM_ALLIN -I${PSCOM_ALLIN_DIR}/ -I${PSCOM_ALLIN_DIR}/../../include/ -I${PSCOM_ALLIN_DIR}/../../lib/pscom/"
 
 	if test -x "${PSCOM_ALLIN_DIR}/../../scripts/vcversion" ; then
